### DISCUSSION TITLE
Filtering a DataGrid on other column values

### DISCUSSION THREAD

**Discussion by tacoWanneeJama** [Megabit/Blazorise#3751] [Q&A] [answered]
For a DataGrid I have a context with an integer column and a string that belong to each other ({34, "34 AL"}, {120, "120 VL"} etc).
I want to show the string column on the screen, but have it order on on the integer column (34 before the 120).
For a solution, I was trying to do in the DataGridColumn: CustomFilter="@((e,s)=>return OnVanNameFilter(e,s, context))", but that gives a compile problem in the _razor.g.cs file.
Do I need to fall back on the CustomFilter of the grid? Or is there another neat way to achieve this?

**Comment by stsrki** [comment]
It's hard to understand. Are you using a tuple or a dictionary?

In either case, having a small sample code to reproduce it would be great to have as that way we would help you faster.

**Comment by tacoWanneeJama** [comment, marked as answer]
Here is the example code:
```
@page "/"

<h3>OrderGrid</h3>

<DataGrid TItem="Van" Data="vans" Filterable=true Sortable=true>
    <DataGridColumns>
        <DataGridColumn Field="VanNr" Caption="Van" >
            <DisplayTemplate>
                @context.VanName
            </DisplayTemplate>
        </DataGridColumn>
        <DataGridColumn Field="VanName" Caption="Van" ></DataGridColumn>
    </DataGridColumns>
</DataGrid>


@code {
    List<Van> vans = new List<Van>()
    {
        new Van(){ VanNr=34, VanName="34 AL"},
        new Van(){ VanNr=102, VanName="102 VL"},
        new Van(){ VanNr=15, VanName="15"}
    };

    public class Van
    {
        public int VanNr { get; set; }  
        public string VanName { get; set; }
    }
}
```
The first column does order correctly, but filtering on text does NOT work.
The second column is filtering OK, but order is not what we want.

**Reply by tacoWanneeJama** [reply]
Does someone have a tip to achieve this?

**Reply by stsrki** [reply]
SO the problem is not filtering but the ordering? You can use `SortField `for that and it will be used for sorting, while `Field `will be used for display.

**Reply by tacoWanneeJama** [reply]
Ah, that does the trick. Thanx
